src/core/qgsweakrelation.h

```diff
@@ -0,0 +1,113 @@
+/***************************************************************************
+  qgsweakrelation.h - QgsWeakRelation
+
+ ---------------------
+ begin                : 5.12.2019
+ copyright            : (C) 2019 by Alessandro Pasotti
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSWEAKRELATION_H
+#define QGSWEAKRELATION_H
+
+#define SIP_NO_FILE
+
+#include "qgis_core.h"
+#include "qgsrelation.h"
+#include "qgsvectorlayerref.h"
+
+/**
+ * The QgsWeakRelation class represent a QgsRelation with possibly
+ * unresolved layers or unmatched fields.
+ *
+ * This class is used to store relation information attached to a
+ * layer style, a method to attempt relation resolution is also
+ * implemented and can be used to create a QgsRelation after the
+ * dependent layers are loaded and available.
+ *
+ * \note not available in Python bindings
+ * \ingroup core
+ * \since QGIS 3.12
+ */
+class CORE_EXPORT QgsWeakRelation
+{
+  public:
+
+    /**
+     * Creates a QgsWeakRelation
+     * \param relationId relation ID
+     * \param relationName relation name
+     * \param strength relation strength
+     * \param referencingLayerId ID of the referencing layer
+     * \param referencingLayerName name of the referencing layer
+     * \param referencingLayerSource source of the referencing layer
+     * \param referencingLayerProviderKey provider name of the referencing layer
+     * \param referencedLayerId ID of the referenced layer
+     * \param referencedLayerName name of the referenced layer
+     * \param referencedLayerSource name of the referenced layer
+     * \param referencedLayerProviderKey provider name of the referenced layer
+     * \param fieldPairs list of the relation field pairs
```

Comment:
I'd propose to remove all those "stating the obvious" param descriptions and `\a` the non obvious / important ones in the inline comment.

Reply:
IMO they are all so obvious, when I wrote this doxy I was thinking: what a waste of time!

Reply:
Kill 'em!

Reply:
Would code-layout agree?

Reply:
If you kill all of them, yes.